Foreign key constraint violation upon deleting organization with group

### Subject of the issue
Unable to delete organizations containing groups (with the groups beta feature enabled).

### Deployment environment
* vaultwarden version: 1.27.0-1ba8275d (HEAD)

* Install method: Built from source

* Clients used: web vault

* Reverse proxy and version: nginx (1.18.0-6.1+deb11u3)

* MySQL/MariaDB or PostgreSQL version: postgres-13 (13.9-0+deb11u1)

* Environment variables:
  ```
  ADMIN_TOKEN=...
  DATABASE_URL=...
  DISABLE_2FA_REMEMBER=false
  DISABLE_ADMIN_TOKEN=true
  DOMAIN=...
  LOGIN_RATELIMIT_MAX_BURST=10
  LOGIN_RATELIMIT_SECONDS=60
  LOG_LEVEL=info
  LOG_TIMESTAMP_FORMAT=''
  ORG_CREATION_USERS=all
  ORG_GROUPS_ENABLED=true
  PASSWORD_HINTS_ALLOWED=false
  REQUIRE_DEVICE_EMAIL=true
  SENDS_ALLOWED=true
  SIGNUPS_ALLOWED=true
  SIGNUPS_DOMAINS_WHITELIST=...
  SIGNUPS_VERIFY=true
  SIGNUPS_VERIFY_RESEND_TIME=300
  SMTP_FROM=...
  SMTP_FROM_NAME=...
  SMTP_HOST=...
  SMTP_PASSWORD=...
  SMTP_PORT=25
  SMTP_SECURITY=starttls
  SMTP_USERNAME=...
  TRASH_AUTO_DELETE_DAYS=30
  USER_ATTACHMENT_LIMIT=131072
  WEBSOCKET_ENABLED=true
  WEBSOCKET_PORT=127.0.0.1
  ```
### Steps to reproduce
1. Create a new organization.
2. Create a group within the new organization.
3. Try to remove the organization. Nothing happens.
4. Try to remove it via admin interface, same thing.

### Expected behaviour
Organization removed.

### Actual behaviour
Organization not removed.

### Troubleshooting data
Related log entries in systemd journal:
```
vaultwarden[3206]: [][request][INFO] DELETE /api/organizations/fbeabd5b-3d65-4b29-9881-37ba2a15d270
postgres[3226]: [9-1] 1676417219.784 [3226] vaultwarden@vaultwarden ERROR:  update or delete on table "organizations" violates foreign key constraint "groups_organizations_uuid_fkey" on table "groups"
postgres[3226]: [9-2] 1676417219.784 [3226] vaultwarden@vaultwarden DETAIL:  Key (uuid)=(fbeabd5b-3d65-4b29-9881-37ba2a15d270) is still referenced from table "groups".
postgres[3226]: [9-3] 1676417219.784 [3226] vaultwarden@vaultwarden STATEMENT:  DELETE  FROM "organizations" WHERE ("organizations"."uuid" = $1) 
vaultwarden[3206]: [][error][ERROR] Error saving organization.
vaultwarden[3206]: [CAUSE] DatabaseError(
vaultwarden[3206]:     ForeignKeyViolation,
vaultwarden[3206]:     "update or delete on table \"organizations\" violates foreign key constraint \"groups_organizations_uuid_fkey\" on table \"groups\"",
vaultwarden[3206]: )
vaultwarden[3206]: [][response][INFO] (delete_organization) DELETE /api/organizations/<org_id> => 400 Bad Request
```

--- a/src/db/models/group.rs
+++ b/src/db/models/group.rs
@@ -151,6 +151,13 @@ impl Group {
         }
     }
 
+    pub async fn delete_all_by_organization(org_uuid: &str, conn: &mut DbConn) -> EmptyResult {
+        for group in Self::find_by_organization(org_uuid, conn).await {
+            group.delete(conn).await?;
+        }
+        Ok(())
+    }
+
     pub async fn find_by_organization(organizations_uuid: &str, conn: &mut DbConn) -> Vec<Self> {
         db_run! { conn: {
             groups::table

--- a/src/db/models/organization.rs
+++ b/src/db/models/organization.rs
@@ -2,7 +2,7 @@ use num_traits::FromPrimitive;
 use serde_json::Value;
 use std::cmp::Ordering;
 
-use super::{CollectionUser, GroupUser, OrgPolicy, OrgPolicyType, TwoFactor, User};
+use super::{CollectionUser, Group, GroupUser, OrgPolicy, OrgPolicyType, TwoFactor, User};
 use crate::CONFIG;
 
 db_object! {
@@ -267,6 +267,7 @@ impl Organization {
         Collection::delete_all_by_organization(&self.uuid, conn).await?;
         UserOrganization::delete_all_by_organization(&self.uuid, conn).await?;
         OrgPolicy::delete_all_by_organization(&self.uuid, conn).await?;
+        Group::delete_all_by_organization(&self.uuid, conn).await?;
 
         db_run! { conn: {
             diesel::delete(organizations::table.filter(organizations::uuid.eq(self.uuid)))

